oil_lang/expr_eval.py

```diff
@@ -579,64 +633,243 @@ def _EvalUnary(self, node):
 
     raise NotImplementedError(node.op.id)
 
-  def _EvalBinary(self, node):
-    # type: (expr__Binary) -> Any # XXX
+  def _ArithNumeric(self, left, right, op):
+    # type: (value_t, value_t, Id_t) -> value_t
+    left = self._MaybeConvertStr(left)
+    right = self._MaybeConvertStr(right)
+    UP_left = left
+    UP_right = right
+
+    with tagswitch(left) as lcase:
+      if lcase(value_e.Int):
+        left = cast(value__Int, UP_left)
+
+        with tagswitch(right) as rcase:
+          if rcase(value_e.Int):
+            right = cast(value__Int, UP_right)
+
+            if op == Id.Arith_Plus:
+              return value.Int(left.i + right.i)
+            elif op == Id.Arith_Minus:
+              return value.Int(left.i - right.i)
+            elif op == Id.Arith_Star:
+              return value.Int(left.i * right.i)
+            elif op == Id.Arith_Slash:
+              if right.i == 0:
+                raise ZeroDivisionError()
+
+              return value.Float(left.i / float(right.i))
+            else:
+              raise NotImplementedError(op)
+
+          elif rcase(value_e.Float):
+            right = cast(value__Float, UP_right)
+            if op == Id.Arith_Plus:
+              return value.Float(left.i + right.f)
```

Comment:
I think we might want to revisit if Int + Float and Int - Float work

Python allows it, and maybe the spec tests did, but we can think about it more, and tighten up the behavior later

Another thing to remember is that we want to do the overflow detection for integers, so it will turn into something like `oils_number_lib.Add(left_int, right_int)`